neo4j/dbtype/vector.go

```diff
@@ -17,10 +17,94 @@
 
 package dbtype
 
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+)
+
 // VectorElement represents the supported element types for Vector.
 type VectorElement interface {
 	~float64 | ~float32 | ~int8 | ~int16 | ~int32 | ~int64
 }
 
 // Vector represents a fixed-length array of numeric values.
 type Vector[T VectorElement] []T
+
+// String returns the string representation of this Vector in the format:
+// vector([data], length, type NOT NULL)
+func (v Vector[T]) String() string {
+	dataStr := formatVectorData(v)
+	length := len(v)
+	typeStr := getVectorTypeString[T]()
+
+	return fmt.Sprintf("vector([%s], %d, %s)", dataStr, length, typeStr)
+}
+
+func getVectorTypeString[T VectorElement]() string {
+	var typeDiscriminator T
+	switch any(typeDiscriminator).(type) {
+	case int8:
+		return "INTEGER8 NOT NULL"
+	case int16:
+		return "INTEGER16 NOT NULL"
+	case int32:
+		return "INTEGER32 NOT NULL"
+	case int64:
+		return "INTEGER NOT NULL"
+	case float32:
+		return "FLOAT32 NOT NULL"
+	case float64:
+		return "FLOAT NOT NULL"
+	default:
+		return "UNKNOWN NOT NULL"
+	}
+}
+
+func formatVectorData[T VectorElement](v Vector[T]) string {
+	if len(v) == 0 {
+		return ""
+	}
+
+	parts := make([]string, len(v))
+	for i, element := range v {
+		parts[i] = formatElement(element)
+	}
+	return strings.Join(parts, ", ")
+}
+
+func formatElement[T VectorElement](element T) string {
+	switch e := any(element).(type) {
+	case float32:
+		return formatFloat(float64(e))
+	case float64:
+		return formatFloat(e)
+	case int8, int16, int32, int64:
+		return fmt.Sprintf("%d", e)
+	default:
+		return fmt.Sprintf("%v", e)
+	}
+}
+
+func formatFloat(f float64) string {
+	if math.IsNaN(f) {
+		return "NaN"
+	}
+	if math.IsInf(f, 1) {
+		return "Infinity"
+	}
+	if math.IsInf(f, -1) {
+		return "-Infinity"
+	}
+	if isWholeNumber(f) {
+		// Ensure we show at least one decimal place
+		return strconv.FormatFloat(f, 'f', 1, 64)
+	}
+	return strconv.FormatFloat(f, 'g', -1, 64)
+}
+
+func isWholeNumber(f float64) bool {
+	const epsilon = 1e-9
+	return math.Abs(f-math.Trunc(f)) < epsilon
```

Comment:
I don't think this is ideal.

```go
{"oh noes", Vector[float64]{1e-50}, "vector([1e-50], 1, FLOAT NOT NULL)"},
```

It shouldn't pretend small floats are zero when they're not in reality.

Are there cases where this wouldn't work?

 ```suggestion
	return f == math.Trunc(f)
```

Reply:
Cheers, went around in a loop thinking `return f == math.Trunc(f)` might have some caveat, but can't find one. Happy to go with this for sure 👍